### PR TITLE
Pptp 1550 Amend contact details

### PR DESCRIPTION
**Pull request by tonymccrae-ee**
### Description of Work carried through

Contact name, email and phone number cab be amended for existing partners.

Screenshot from 2022-01-25 13-11-16

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [x] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [ ] User Acceptance Tests (UAT) were run locally and have passed.
 - [ ] No Accessibility errors/warnings reported by Wave

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partner/PartnerContactNameController.scala [hunk 1]
```diff
@@ -49,64 +52,108 @@ class PartnerContactNameController @Inject() (
   def displayPage(): Action[AnyContent] =
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partner/PartnerContactNameController.scala:52]: Perhaps rename to `displayNewPartner()` for consistency with the amend equivalent?

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partner/PartnerContactNameController.scala [hunk 2]
```diff
@@ -49,64 +52,108 @@ class PartnerContactNameController @Inject() (
...
+      case Some(data) =>
+        MemberName.form().fill(MemberName(data._1, data._2))
+      case None =>
+        MemberName.form()
+    }
+
+    Ok(page(form, partner.name, backCall, submitCall))
+  }
+
   def submit(): Action[AnyContent] =
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partner/PartnerContactNameController.scala:91]: `submitNewPartner()` for consistency?

File: build.sbt [hunk 3]
```diff
@@ -48,10 +48,6 @@ lazy val scoverageSettings: Seq[Setting[_]] = Seq(
     "test\\..*",
     ".*(BuildInfo|Routes|Options).*",
     "logger.*\\(.*\\)",
-    "uk.gov.hmrc.plasticpackagingtax.registration.controllers.partnership.PartnershipOtherPartnerCheckAnswersController",
-    "uk.gov.hmrc.plasticpackagingtax.registration.controllers.partnership.PartnershipPartnersListController",
-    "uk.gov.hmrc.plasticpackagingtax.registration.views.html.partnerships.other_partners_page",
-    "uk.gov.hmrc.plasticpackagingtax.registration.views.html.partnerships.check_other_partner_answers_page"
```
Comment (RyanKellyHMRC) [inline build.sbt:54]: ⭐

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partner/PartnerContactNameController.scala [hunk 4]
```diff
@@ -49,64 +52,108 @@ class PartnerContactNameController @Inject() (
...
+  def submitExistingPartner(partnerId: String): Action[AnyContent] =
+    (authenticate andThen journeyAction).async { implicit request =>
+      def updateAction(memberName: MemberName): Future[Either[ServiceError, Registration]] =
+        updateExistingPartner(memberName, partnerId)
+
+      request.registration.findPartner(partnerId).map { partner =>
+        handleSubmission(partner,
+                         partnerRoutes.PartnerCheckAnswersController.displayExistingPartner(
+                           partnerId
+                         ),
```
[the remark this answers is not in the document]
Comment (tonymccrae-ee) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partner/PartnerContactNameController.scala:115]: Yeah... this dropped out unexpected easily and felt better.
No big to rewire to got down the whole flow if that's what we decide.

**Review by RyanKellyHMRC — APPROVED** [review]
Super

**Pull request merged**